src/Support/Metrics/Trend.php

```diff
@@ -0,0 +1,215 @@
+            },
+            'pgsql' => match ($this->unit) {
+                'year' => "to_char($dateColumn, 'YYYY')",
+                'month' => "to_char($dateColumn, 'YYYY-MM')",
+                'week' => "to_char($dateColumn, 'IYYY-IW')",
+                'day' => "to_char($dateColumn, 'YYYY-MM-DD')",
+                'hour' => "to_char($dateColumn, 'YYYY-MM-DD HH24:00')",
+                'minute' => "to_char($dateColumn, 'YYYY-MM-DD HH24:mi:00')",
+            },
+            'sqlsrv' => match ($this->unit) {
+                'year' => "FORMAT($dateColumn, 'yyyy')",
+                'month' => "FORMAT($dateColumn, 'yyyy-MM')",
+                'week' => "concat(YEAR($dateColumn), '-', datepart(ISO_WEEK, $dateColumn))",
+                'day' => "FORMAT($dateColumn, 'yyyy-MM-dd')",
+                'hour' => "FORMAT($dateColumn, 'yyyy-MM-dd HH:00')",
+                'minute' => "FORMAT($dateColumn, 'yyyy-MM-dd HH:mm:00')",
+            },
+            default => throw new InvalidArgumentException('Laravel Easy Metrics is not supported for this database.')
+        };
+    }
+
+    protected function getFormat(): string
+    {
+        return match ($this->unit) {
+            'year' => 'Y',
+            'month' => 'Y-m',
+            'week' => 'Y-W',
+            'day' => 'Y-m-d',
+            'hour' => 'Y-m-d H:00',
+            'minute' => 'Y-m-d H:i:00',
+            default => throw new InvalidArgumentException('Invalid unit: ' . $this->unit),
+        };
+    }
+
+    protected function getStartingDate(): CarbonImmutable
+    {
+        if ($this->getRange() instanceof TimeFrameEnum && $timeFrameRange = $this->getRange()?->getRange()) {
+            return $timeFrameRange[0];
+        }
+
+        if ($this->startingDate && $this->getRange() === TimeFrameEnum::Custom) {
+            return $this->startingDate;
+        }
+
+        $now = CarbonImmutable::now();
+        $range = $this->getRange() - 1;
+
+        return match ($this->unit) {
+            'year' => $now
+                ->subYearsWithoutOverflow($range)
+                ->firstOfYear()
+                ->setTime(0, 0),
+            'month' => $now
+                ->subMonthsWithoutOverflow($range)
+                ->firstOfMonth()
+                ->setTime(0, 0),
+            'week' => $now
+                ->subWeeks($range)
+                ->startOfWeek()
+                ->setTime(0, 0),
+            'day' => $now
+                ->subDays($range)
+                ->setTime(0, 0),
+            'hour' => $now
+                ->subHours($range),
+            'minute' => $now
+                ->subMinutes($range),
+            default => throw new InvalidArgumentException('Invalid unit: ' . $this->unit),
+        };
+    }
+
+    protected function getEndingDate(): CarbonImmutable
+    {
+        if ($this->getRange() instanceof TimeFrameEnum && $timeFrameRange = $this->getRange()?->getRange()) {
```

Comment:
no need to make it null safe $this->getRange()?->getRange(). At this point $this->getRange() can not be null.